flink-end-to-end-tests/flink-end-to-end-tests-common-kafka/src/test/java/org/apache/flink/tests/util/kafka/KafkaSourceE2ECase.java

```diff
@@ -0,0 +1,88 @@
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.tests.util.kafka;
+
+import org.apache.flink.connector.kafka.source.testutils.KafkaMultipleTopicExternalContext;
+import org.apache.flink.connector.kafka.source.testutils.KafkaSingleTopicExternalContext;
+import org.apache.flink.connectors.test.common.external.DefaultContainerizedExternalSystem;
+import org.apache.flink.connectors.test.common.junit.annotations.ExternalContextFactory;
+import org.apache.flink.connectors.test.common.junit.annotations.ExternalSystem;
+import org.apache.flink.connectors.test.common.junit.annotations.TestEnv;
+import org.apache.flink.connectors.test.common.testsuites.SourceTestSuiteBase;
+import org.apache.flink.tests.util.TestUtils;
+import org.apache.flink.tests.util.flink.FlinkContainerTestEnvironment;
+
+import org.testcontainers.containers.KafkaContainer;
+import org.testcontainers.utility.DockerImageName;
+
+/** Kafka E2E test based on connector testing framework. */
+public class KafkaSourceE2ECase extends SourceTestSuiteBase<String> {
+    private static final String KAFKA_HOSTNAME = "kafka";
+    private static final String KAFKA_IMAGE_NAME = "confluentinc/cp-kafka:5.5.2";
+
+    // Defines TestEnvironment
+    @TestEnv FlinkContainerTestEnvironment flink;
+
+    // Defines ConnectorExternalSystem
+    @ExternalSystem DefaultContainerizedExternalSystem<KafkaContainer> kafka;
+
+    // Defines 2 External context Factories, so test cases will be invoked twice using these two
+    // kinds of external contexts.
+    @ExternalContextFactory KafkaSingleTopicExternalContext.Factory singleTopic;
+    @ExternalContextFactory KafkaMultipleTopicExternalContext.Factory multipleTopic;
+
+    /** Instantiate and preparing test resources. */
+    public KafkaSourceE2ECase() throws Exception {
+
+        // Flink on containers
+        flink =
+                new FlinkContainerTestEnvironment(
+                        1,
+                        6,
+                        TestUtils.getResource("kafka-connector.jar").toAbsolutePath().toString(),
+                        TestUtils.getResource("kafka-clients.jar").toAbsolutePath().toString());
+
+        // Kafka on container has to be bound with Flink containers so that they can access each
+        // other through internal network, so we have to launch Flink containers here.
+        flink.startUp();
```

Comment:
This should be unnecessary. Isn't the dependsOn in `Builder#bindWithFlinkContainer` enough?